### ISSUE TITLE
Introspection

### ISSUE THREAD

**Issue by melinath** [melinath/django-graph-api#33]
Add full schema introspection support so that GraphiQL is more useful.

**Comment by ariannedee** [comment]
Maybe I prematurely approved #45, but here is a list that I think need to be addressed before closing #33.

1. For the query
```
{
  __schema {
    types {
      fields {
        type {
          name 
        }
     }
  }
}
```
Type -> name is not returning the expected result. For List types, name should return null, but it currently causes a server error. For Object types, it currently returns 'Object', but I think it should return 'Character'.

2. For the query
```
{
  __schema {
    types {
      fields {
        type {
          fields {
            name
          }
        }
     }
  }
}
```
The nested 'fields' selection always returns an empty list.

3. List types should have an `ofType` field.

4. 'directives' should return an empty list

**Comment by ariannedee** [comment]
Fixed in #51
